### ISSUE TITLE
Azure compute_vm_scaleset reports events with error.message when there's no vm scaleset

### ISSUE THREAD

**Issue by kaiyan-sheng** [elastic/beats#14317]
When testing azure metricbeat module, I don't have a vm scale set created in my subscription. But with `compute_vm_scaleset` enabled in azure.yml, I got two events reported to Elasticsearch with `error.message` in one collection period: 

#### event1:
```
{
  "_index": "metricbeat-8.0.0-2019.10.24-000001",
  "_type": "_doc",
  "_id": "Ac69GW4Byjz_HC7wQRow",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2019-10-29T22:57:42.435Z",
    "agent": {
      "id": "23fd6ee9-2255-43f3-9565-78a051337c04",
      "version": "8.0.0",
      "type": "metricbeat",
      "ephemeral_id": "4c43aaf9-5ef2-42c8-a5f8-5c4d4130b98d",
      "hostname": "KaiyanMacBookPro"
    },
    "event": {
      "dataset": "azure.compute_vm_scaleset",
      "module": "azure",
      "duration": 316808318
    },
    "metricset": {
      "name": "compute_vm_scaleset",
      "period": 300000
    },
    "service": {
      "type": "azure"
    },
    "error": {
      "message": "failed to retrieve resources: No resources returned using the configuration options resource ID [], resource group [], resource type , resource query resourceType eq 'Microsoft.Compute/virtualMachineScaleSets'"
    },
    "ecs": {
      "version": "1.2.0"
    },
    "host": {
      "name": "KaiyanMacBookPro",
      "id": "9C7FAB7B-29D1-5926-8E84-158A9CA3E25D",
      "hostname": "KaiyanMacBookPro",
      "architecture": "x86_64",
      "os": {
        "version": "10.13.6",
        "family": "darwin",
        "name": "Mac OS X",
        "kernel": "17.7.0",
        "build": "17G7024",
        "platform": "darwin"
      }
    }
  },
  "fields": {
    "@timestamp": [
      "2019-10-29T22:57:42.435Z"
    ]
  },
  "sort": [
    1572389862435
  ]
}
```

#### event2:
```
{
  "_index": "metricbeat-8.0.0-2019.10.24-000001",
  "_type": "_doc",
  "_id": "As69GW4Byjz_HC7wQRow",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2019-10-29T22:57:42.435Z",
    "service": {
      "type": "azure"
    },
    "error": {
      "message": "no resources were found based on all the configurations options entered"
    },
    "event": {
      "dataset": "azure.compute_vm_scaleset",
      "module": "azure",
      "duration": 316825806
    },
    "ecs": {
      "version": "1.2.0"
    },
    "host": {
      "id": "9C7FAB7B-29D1-5926-8E84-158A9CA3E25D",
      "hostname": "KaiyanMacBookPro",
      "architecture": "x86_64",
      "name": "KaiyanMacBookPro",
      "os": {
        "family": "darwin",
        "name": "Mac OS X",
        "kernel": "17.7.0",
        "build": "17G7024",
        "platform": "darwin",
        "version": "10.13.6"
      }
    },
    "agent": {
      "id": "23fd6ee9-2255-43f3-9565-78a051337c04",
      "version": "8.0.0",
      "type": "metricbeat",
      "ephemeral_id": "4c43aaf9-5ef2-42c8-a5f8-5c4d4130b98d",
      "hostname": "KaiyanMacBookPro"
    },
    "metricset": {
      "period": 300000,
      "name": "compute_vm_scaleset"
    }
  },
  "fields": {
    "@timestamp": [
      "2019-10-29T22:57:42.435Z"
    ]
  },
  "sort": [
    1572389862435
  ]
}
```

I don't think it's necessary to create an empty event for this since there is no actual vm scale set exists in the environment. Also I'm not sure why there are two events created by `compute_vm_scaleset` in the same collection period in this case?

@narph WDYT?

**Comment by kaiyan-sheng** [comment]
Same thing also happened for `monitor` scaleset:
```
{
  "_index": "metricbeat-8.0.0-2019.10.24-000001",
  "_type": "_doc",
  "_id": "AM69GW4Byjz_HC7wQRow",
  "_version": 1,
  "_score": null,
  "_source": {
    "@timestamp": "2019-10-29T22:57:42.411Z",
    "error": {
      "message": "no resources were found based on all the configurations options entered"
    },
    "event": {
      "duration": 308964161,
      "dataset": "azure.monitor",
      "module": "azure"
    },
    "metricset": {
      "name": "monitor",
      "period": 300000
    },
    "ecs": {
      "version": "1.2.0"
    },
    "host": {
      "id": "9C7FAB7B-29D1-5926-8E84-158A9CA3E25D",
      "name": "KaiyanMacBookPro",
      "hostname": "KaiyanMacBookPro",
      "architecture": "x86_64",
      "os": {
        "platform": "darwin",
        "version": "10.13.6",
        "family": "darwin",
        "name": "Mac OS X",
        "kernel": "17.7.0",
        "build": "17G7024"
      }
    },
    "agent": {
      "hostname": "KaiyanMacBookPro",
      "id": "23fd6ee9-2255-43f3-9565-78a051337c04",
      "version": "8.0.0",
      "type": "metricbeat",
      "ephemeral_id": "4c43aaf9-5ef2-42c8-a5f8-5c4d4130b98d"
    },
    "service": {
      "type": "azure"
    }
  },
  "fields": {
    "@timestamp": [
      "2019-10-29T22:57:42.411Z"
    ]
  },
  "sort": [
    1572389862411
  ]
}
```

**Comment by narph** [comment]
@kaiyan-sheng , yes, at the moment, when no resources are found at the moment we also report in es. 
I have chosen to go this route as users can create queries trying to identify specific resources and those can be either removed or shutdown/unavailable and they might want to see those type of events. @exekias let me know what you think about this?

Regarding the double events in the compute_vm_scaleset, you are saying the same thing happens in the monitor metricset, are you getting double events in all metricsets? 
Can you share the configuration (without credentials) in order to investigate further?

**Comment by exekias** [comment]
If no error occurred when retrieving this info it should be ok to not report any document. I think users would still have enough tools to know they don't have any scaleset (just doing some counting on the right metric).

I'm +1 to remove these docs and only report errors when we get an actual error response from the API.

In any case I don't think we need to backport this change

**Comment by narph** [comment]
I was able to reproduce the scenario when the 2 events are created in compute_vm_scaleset. The error messages are different and that is because we go over each resource option and if no resources are found then we return a message. After looping through all resource options we check if any resources in total are found so we can continue to the next step, if not, an error message is returned.
I have created https://github.com/elastic/beats/pull/14327 which will make sure these messages are only entered in the logs and no events are created.
thanks @kaiyan-sheng , @exekias  for your input.
